packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -206,14 +218,18 @@ export class Session extends EventEmitter {
                 for (const client of clients) {
                     client.send({ ...message }, {
                         resolve: (payload: Payloads.Payload, response: Message) => {
-                            // Cache the response message.
-                            if (client.authoritative) {
-                                authoritativeReplyMessage = {...response };
-                            } else if (!fallbackReplyMessage) {
-                                fallbackReplyMessage = { ...response };
-                            }
                             // Decrement the wait counter.
                             awaitCount -= 1;

```

Comment:
>awaitCount -= 1; [](start = 28, length = 16)

What happens when a client disconnects during this process? It might be better to see if the number of responses is equal to the number of connected clients or something.

Reply:
When a client disconnects the reject handler (below) is called and the awaitCount decremented there.

---
In reply to: [256942555](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256942555) [](ancestors = 256942555)

Reply:
If that was changed then there could the opposite problem - if a disconnecting client has already received the response. 
Does disconnecting trigger a response? if so, then Eric's code would always reach 0

---
In reply to: [256942555](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256942555) [](ancestors = 256942555)

Reply:
Eric answering my question before I even asked it :-)

---
In reply to: [256943726](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256943726) [](ancestors = 256943726,256942555)

Reply:
It's a tricky piece of code for sure.

---
In reply to: [256943945](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/192#discussion_r256943945) [](ancestors = 256943945,256943726,256942555)